eva-server/src/test/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2IntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+        assertEquals(0, response.getBody().getResponse().size());
+    }
+
+    @Test
+    public void rootTestForError() throws URISyntaxException {
+        String url = "/v2/variants/13:32889669:C:T?species=&assembly=grcm38";
+        assertEquals("Please specify a species", testForErrorHelper(url));
+    }
+
+    private String testForErrorHelper(String url) {
+        return WSTestHelpers.testRestTemplateHelperForError(url, restTemplate);
+    }
+
+    @Test
+    public void annotationEndPointTestExisting() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T/annotations?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
+        assertFalse(annotations.getBody().getResponse().get(0).getResult().get(0).getChromosome().isEmpty());
+    }
+
+    @Test
+    public void annotationEndPointTestNonExisting() throws URISyntaxException {
+        String url = "/v2/variants/100:0:C:T/annotations?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<Annotation>>> annotations = restTemplate.exchange(
+                url, HttpMethod.GET, null,
+                new ParameterizedTypeReference<QueryResponse<QueryResult<Annotation>>>() {
+                });
+        assertEquals(HttpStatus.NOT_FOUND, annotations.getStatusCode());
+        assertTrue(annotations.getBody().getResponse().get(0).getResult().size() == 0);
+    }
+
+    @Test
+    public void annotationEndpointTestForError() throws URISyntaxException {
+        String url = "/v2/variants/13:32889669:C:T/annotations?species=mmusculus&assembly=grcm38&" +
+                "annot-vep-version=1";
+        assertEquals("Please specify either both annotation VEP version and annotation VEP cache version, " +
+                "or neither", testForErrorHelper(url));
+    }
+
+    @Test
+    public void sourceEntriesEndPointTestExisting() throws URISyntaxException {
+        String url = "/v2/variants/20:60100:A:T/source-entries?species=mmusculus&assembly=grcm38";
+        ResponseEntity<QueryResponse<QueryResult<VariantSourceEntryWithSampleNames>>> annotations = restTemplate.
+                exchange(url, HttpMethod.GET, null,
+                        new ParameterizedTypeReference<QueryResponse<QueryResult
+                                <VariantSourceEntryWithSampleNames>>>() {
+                        });
+        assertEquals(HttpStatus.OK, annotations.getStatusCode());
+        assertFalse(annotations.getBody().getResponse().get(0).getResult().get(0).getFileId().isEmpty());
+    }
+
+    @Test
+    public void sourceEntriesEndPointTestNonExisting() throws URISyntaxException {
```

Comment:
Same as in the method for testing annotations, it should check the case where the variant doesn't exist, or when it exists and the statistics don't.

Reply:
I have added this :)